external-crates/move/crates/move-compiler/src/sui_mode/typing.rs

```diff
@@ -673,6 +682,24 @@ fn is_mut_clock(param_ty: &Type) -> bool {
     }
 }
 
+fn is_mut_random(param_ty: &Type) -> bool {
+    match &param_ty.value {
+        Type_::Ref(/* mut */ false, _) => false,
+        Type_::Ref(/* mut */ true, t) => is_mut_random(t),
+        Type_::Apply(_, sp!(_, n_), _) => n_.is(
+            SUI_ADDR_NAME,
+            RANDOMNESS_MODULE_NAME,
+            RANDOMNESS_STATE_TYPE_NAME,
+        ),
+        Type_::Unit
+        | Type_::Param(_)
+        | Type_::Var(_)
+        | Type_::Anything
+        | Type_::UnresolvedError
+        | Type_::Fun(_, _) => false,
```

Comment:
Same question about function signatures

Reply:
only arguments (aligned with what we do for `Clock`)